src/types/routeDef.ts

```diff
@@ -23,6 +23,21 @@ export type RouteItemDef = {
    * This route can not be access after logging in
    */
   isAuthRoute?: boolean;
+  /** Navigation title of menu item for navbar or sidebar */
+  navigationTitle?: string;
+  /** Page title of the screen to be shown on the header */
+  pageTitle?: string;
+  /** Icon of menu item for sidebar using https://material.io/resources/icons */
+  icon?: any; // TODO: add the correct Type for the material icon
```

Comment:
@themitvp I need help to figure out what should be the correct type for the icon element. I tried ReactElement and ComponentType but nothing worked for me

Reply:
We can do two things:
1. `React.ReactNode`: This will allow any component or array of components to be added.
2. `OverridableComponent<SvgIconTypeMap>`: They come from material-ui, and by doing so we only allow material-ui icons to be used. 

The advantage for 1. is that we can add special components if we need to, but the advantage for 2. is that we are saying that we only expect material ui icons, and the compiler will complain if you are doing something else.

So maybe best that we go with 2. and then see if we end of in a situation where we need to change it?

Reply:
Thank you @themitvp for the help,` OverridableComponent<SvgIconTypeMap>` works perfectly 🙏